infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+
+@Slf4j
+// TODO (Chetan): Remove this after completion
+@Ignore("Still under development.")
+public class CorfuSequencerServerTest {
+
+    private final AtomicInteger requestCounter = new AtomicInteger();
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+    @Mock
+    Map<UUID, StreamAddressSpace> mockStreamAddressSpaceMap;
+    // The SequencerServer instance used for testing
+    private SequencerServer sequencerServer;
+    // Objects that need to be mocked
+    @Mock
+    private ServerContext mockServerContext;
+    @Mock
+    private IServerRouter mockServerRouter;
+    @Mock
+    private ChannelHandlerContext mockChannelHandlerContext;
+    @Spy
+    private SequencerServer.FactoryHelper spyFactoryHelper;
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId indicates if the message is clusterId aware
+     * @param ignoreEpoch     indicates if the message is epoch aware
+     * @return the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * A helper method that compares the base fields of two message headers.
+     * These include the request ID, the epoch, the client ID, and the cluster ID.
+     *
+     * @param requestHeader  the header from the request message
+     * @param responseHeader the header from the response message
+     * @return true if the two headers have the same base field values
+     */
+    private boolean compareBaseHeaderFields(HeaderMsg requestHeader, HeaderMsg responseHeader) {
+        return requestHeader.getRequestId() == responseHeader.getRequestId() &&
+                requestHeader.getEpoch() == responseHeader.getEpoch() &&
+                requestHeader.getClientId().equals(responseHeader.getClientId()) &&
+                requestHeader.getClusterId().equals(responseHeader.getClusterId());
+    }
+
+    /**
+     * Perform the required preparation before running individual tests.
+     * This includes preparing the mocks and initializing the DirectExecutorService.
+     */
+    @Before
+    public void setup() {
+        // Initialize with newDirectExecutorService to execute the server RPC
+        // handler methods on the calling thread
+        when(mockServerContext.getExecutorService(anyInt(), anyString()))
+                .thenReturn(MoreExecutors.newDirectExecutorService());
+    }
+
+    /**
+     * Test that the SequencerServer sends NotReadyErrorMsg when client sends a TokenRequestMsg
+     * without bootstrapping it before.
+     * Essentially tests the {@link SequencerServer#isServerReadyToHandleMsg(RequestMsg)} method.
+     */
+    @Test
+    public void testSequencerServerNotReady() {
```

Comment:
Can we also verify that some of the state was not updated? Methods like `setSequencerEpoch` should not have been invoked in this case.